### PR TITLE
[1LP][RFR] gen_data refactoring to use _creating_skeleton; preserve http response status

### PR DESCRIPTION
**Pull request by mkoura**
* preserve HTTP response status so it can be checked in the test
* use `_creating_skeleton` wherever possible
* refactoring

Tests in files specified below are using all of the methods in `gen_data` apart from `arbitration_settings` and `orchestration_templates` that I tested locally.

{{pytest: cfme/tests/configure/test_rest_access_control.py cfme/tests/configure/test_tag.py -v -k REST --use-provider vsphere6-nested}}

### CODE REVIEW

File: cfme/rest/gen_data.py [hunk 1]
```diff
@@ -193,50 +152,24 @@ def _finished():
...
-
-    @request.addfinalizer
-    def _finished():
-        services = [_ for _ in rest_api.collections.services]
-        if len(services) != 0:
-            rest_api.collections.services.action.delete(*services)
+    new_service = service_data(request, rest_api, a_provider, dialog, service_catalogs)
+    rest_service = rest_api.collections.services.get(name=new_service['service_name'])
+    # tests expect iterable
+    return [rest_service]
```
Comment (mkoura) [inline cfme/rest/gen_data.py]: return and delete (in service_data) just the service that was created here, don't touch anything else

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

File: cfme/rest/gen_data.py [hunk 2]
```diff
@@ -270,36 +203,40 @@ def _finished():
...
+
+    s_tpls = [ent for ent in rest_api.collections.service_templates if ent.name in new_names]
 
     @request.addfinalizer
     def _finished():
-        s_tpls = [_ for _ in rest_api.collections.service_templates]
-        if len(s_tpls) != 0:
+        to_delete = [ent for ent in rest_api.collections.service_templates if ent.name in new_names]
+        if len(to_delete) != 0:
             rest_api.collections.service_templates.action.delete(*s_tpls)
```
Comment (izapolsk) [inline cfme/rest/gen_data.py]: REQUIRED: I suppose you wanted to use to_delete here

File: cfme/rest/gen_data.py [hunk 3]
```diff
@@ -334,15 +271,15 @@ def automation_requests_data(vm, requests_collection=False, approve=True):
...
+    return [data for _ in range(num)]
 
 
 def groups(request, rest_api, role, tenant, num=1):
     data = [{
         "description": "group_description_{}".format(fauxfactory.gen_alphanumeric()),
         "role": {"href": role.href},
         "tenant": {"href": tenant.href}
-    } for index in range(0, num)]
+    } for _ in range(num)]
```
Comment (izapolsk) [inline cfme/rest/gen_data.py]: OPTIONAL: pep doesn't recommend to use list comprehensions if those cannot fit in one line.

File: cfme/rest/gen_data.py [hunk 4]
```diff
@@ -31,83 +29,44 @@
...
+        })
 
-    return ctgs
+    return _creating_skeleton(request, rest_api, 'categories', ctg_data)
 
 
 def tags(request, rest_api, categories):
     # Category id, href or name needs to be specified for creating a new tag resource
     tags = []
     for i, ctg in enumerate(categories):
```
Comment (izapolsk) [inline cfme/rest/gen_data.py]: OPTIONAL: don't like one letter variables.

**Comment by mkoura** [comment]
@psav I can confirm that all the deletes that were removed are now handled in `_creating_skeleton`.

**Pull request merged**